### PR TITLE
Allow dynamodb health check when credentials are not provided in options

### PR DESCRIPTION
**Pull request by lisabii**
Make the options and the credentials of dynamo db check optional.
The AWS credentials and region in my project were set in the env params, it will be better to allow the user to leave the options empty. Unit test added and fully tested.

### CODE REVIEW

File: src/HealthChecks.DynamoDb/DynamoDbHealthCheck.cs [hunk 1]
```diff
@@ -14,16 +14,25 @@ public class DynamoDbHealthCheck
...
+                var keysProvided = !string.IsNullOrEmpty(_options.AccessKey) &&
+                                   !string.IsNullOrEmpty(_options.SecretKey);
+                var regionProvided = _options.RegionEndpoint == null;
+                
+                if (keysProvided)
+                {
+                    var credentials = new BasicAWSCredentials(_options.AccessKey, _options.SecretKey);
+                    client = regionProvided ? new AmazonDynamoDBClient(credentials, _options.RegionEndpoint) : new AmazonDynamoDBClient(credentials);
+                }
+                else if(regionProvided)
```
Comment (sungam3r) [inline src/HealthChecks.DynamoDb/DynamoDbHealthCheck.cs:32]:
```suggestion
                else if (regionProvided)

```

**Review by albertocorrales — APPROVED** [review]
Changes of this PR look good to me. For example right now I'm working with policies instead of credentials to connect to DynamoDb and this change will allow me to use this healthcheck without credentials.

**Comment by albertocorrales** [comment]
I'm reading this PR and I think it might solve the same problem https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/pull/653

**Comment by sungam3r** [comment]
@lisabii Please rebase onto `master` if you are going to continue work on this PR. Then I can review/merge.

**Comment by sungam3r** [comment]
ping @lisabii

**Comment by sungam3r** [comment]
@lisabii Do you intend to continue working on that PR?

**Comment by sungam3r** [comment]
Superseded by #1870. Thanks, @lisabii .

**Pull request closed without merging**